tests/nat/nat_helpers.py

```diff
@@ -0,0 +1,848 @@
+       "green": {"ip": "12.1.0.2", "id": "3", "mask": "30", "gw": "12.1.0.1", "port_id": "29", "dut_iface": "PortChannel0002"},
+       "yellow": {"ip": "192.168.0.201", "id": "4", "mask": "24", "gw": "192.168.0.1", "port_id": "3"}
+       }
+SETUP_CONF = {"loopback": {"vrf": VRF, "acl_subnet": ACL_SUBNET},
+              "port_in_lag": {"vrf": VRF, "acl_subnet": ACL_SUBNET}
+              }
+DIRECTION_PARAMS = ['host-tor', 'leaf-tor']
+FULL_CONE_TEST_IP = "172.20.1.2"
+FULL_CONE_TEST_SUBNET = "172.20.1.0/24"
+REBOOT_MAP = {'cold': {"timeout": 300}, 'fast': {"timeout": 180}, 'warm': {"timeout": 180}}
+
+
+def configure_nat_over_cli(duthost, flow_direction, action, nat_type, global_ip, local_ip, proto=None,
+                           global_port=None, local_port=None):
+    """
+    static NAT/NAPT CLI wrapper
+    :param duthost: DUT host object
+    :param flow_direction: string traffic direction
+    :param action: string rule action
+    :param nat_type: string static nat type
+    :param global_ip: string global IP address value
+    :param local_ip: string local IP address value
+    :param proto: string protocol type
+    :param global_port: string global l4 port
+    :param local_port: string local l4 port
+    :return : dict with rule parameters
+    """
+    if nat_type == 'static_nat':
+        entry = {
+            global_ip: {'local_ip': local_ip}
+        }
+        duthost.command("sudo config nat {} static basic {} {}".format(action, global_ip, local_ip))
+        return entry
+    elif nat_type == 'static_napt':
+        static_napt_type = DIRECTION_MAP[flow_direction]
+        action_type_map = {'add': '-nat_type {}'.format(static_napt_type), 'remove': ''}
+        entry = {
+            "{}|{}|{}".format(global_ip, proto.upper(), global_port): {'local_ip': local_ip,
+                                                                       'local_port': "{}".format(local_port),
+                                                                       'nat_type': static_napt_type
+                                                                       }
+        }
+        duthost.command("sudo config nat {} static {} {} {} {} {} {}".format(action, proto.lower(),
+                                                                             global_ip, global_port,
+                                                                             local_ip, local_port,
+                                                                             action_type_map[action]))
+        return entry
+
+
+def nat_statistics(duthost, show=False, clear=False):
+    """
+    NAT CLI helper which gets or clears NAT statistics
+    :param duthost: DUT host object
+    :param show: bool
+    :param clear: bool
+    :return : formatted CLI output
+    """
+    if show:
+        output_cli = exec_command(duthost, ["show nat statistics"])
+        if output_cli["rc"]:
```

Comment:
in some place exec_command is used and in some other places duthost.command is used, can we use function to configure the DUT every where

Reply:
In this particular case exec_command was being used because it handle error message in case of negative case scenario:
        try:
            response = host.shell(command_list[0])
            return response
        except Exception as e:
            return e.__str__() 
It will return exception string to analyze